sql/components.sql

```diff
@@ -51,6 +52,122 @@ FROM (
 ) x;
 $$;
 
+create or replace function oas_build_component_schemas_headers()
+returns jsonb language sql as
+$$
+select jsonb_build_object(
+  'header.preferParams',
+  oas_schema_object(
+    type := 'object',
+    properties := jsonb_build_object(
+      'params',
+      oas_schema_object(
+        description := 'Send JSON as a single parameter',
+        type := 'string',
+        "enum" := '["single-object"]'
+      )
+    )
+  ),
```

Comment:
Could we mark this as deprecated?

https://stackoverflow.com/questions/49870109/how-to-annotate-a-field-as-deprecated-in-openapi-swagger-2-0

Reply:
See https://postgrest.org/en/latest/references/api/preferences.html#prefer-params

Reply:
Cool, yeah, it should let us do that.

Reply:
:+1: If it's considerable work it can be left for another PR

Reply:
It's done. Only  adding the `deprecated` property is enough.